apps/07jquery-tic-tac-toe/script.js

```diff
@@ -50,8 +43,23 @@ function checkForTie() {
   }
 }
 
+$('div[data-cell]').click( function() {
+  if ($(this).text() === '' && play) {
+    $(this).text(playerTurn);
+    if (checkForWin()) {
+      $('#announce-winner').text('Player ' + playerTurn + ' Wins!');
+    }
+    playerTurn = (playerTurn === 'X') ? 'O' : 'X';
+  }
```

Comment:
This event listener should throw an error message when the player makes an invalid move.